release notes/v0.40.0.md

```diff
@@ -0,0 +1,253 @@
+
+Current issues and future improvements for the WebSockets API can be found in it's [issue tracker](https://github.com/grafana/xk6-websockets/issues). Currently documentation is available through [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) although some incompatibility apply:
+- no Blob binary type - ArrayBuffer is the default
+- no `onMessage` and co. - only addEventListener
+
+
+## First class support for JavaScript Classes
+
+As part of [updating goja](https://github.com/grafana/k6/pull/2610) k6 got native, as in not by transpiling by the internal babel, support for classes 🎉. Because this actually implements classes as they described in the latest ECMAScript specification this also means we get a ton of additional class features that were never previously supported such as private fields. Additionally at least one bug [#1763](https://github.com/grafana/k6/issues/1763) was fixed as a result of this, but probably many more as well.
+
+Due to this fairly big change there are chances that some code will now behave differently. Please report any issues, though take into consideration that it's possible that the new behavior is just the correct one.
+
+Other updates from goja are:
+- optimizations around using strings and some access patterns.
+- support for `\u{01234}` unicode point syntax in regexp.
+- fixes a case where interrupting the VM would not have worked, especially around try/catch usage. This was particularly problematic for k6 as this could lead to the whole k6 hanging forever. [#2600](https://github.com/grafana/k6/pull/2600)
+
+Many thanks to @dop251 for continuing to improve goja!🎉🙇
+
+### New Test runtime for module extension developers [#2598](https://github.com/grafana/k6/issues/2598)
+
+While we are developing extensions internally it constantly comes up that you need to do create the same structure over and over again when you run stuff in tests. With the addition of the event loops it is now required to setup it as well. Even k6 team members get parts of this wrong every once in a while, so we decided to add a small type to be used by (extension) module developers to write tests easier. [#2602](https://github.com/grafana/k6/pull/2602)
+
+[This API](https://pkg.go.dev/go.k6.io/k6@v0.39.1-0.20220819115243-e09bb8727786/js/modulestest#Runtime) will likely change and evolve as we add more functionality or as we change the k6 internal API.
+
+### `k6/ws` take `throw` option into consideration [#2247](https://github.com/grafana/k6/issues/2247)
+
+`k6/http` has been using the [`throw` option](https://k6.io/docs/using-k6/k6-options/reference/#throw) to figure out if should throw an exception on errors or whether it should return an response object with error set on it (and log it).
+
+This functionality is finally also available for `k6/ws` which previous to this would've *always* thrown an exception making scripting around it more involved.[#2616](https://github.com/grafana/k6/pull/2616)
```

Comment:
```suggestion
This functionality is finally also available for `k6/ws` which previous to this would've *always* thrown an exception making scripting around it more involved. [#2616](https://github.com/grafana/k6/pull/2616)
```